ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/CodeMirror.tsx

```diff
@@ -340,9 +340,26 @@ function CodeMirror(props: NoteBodyEditorProps, ref: any) {
 
 	useEffect(() => {
 		if (props.searchMarkers !== previousSearchMarkers || renderedBody !== previousRenderedBody) {
-			webviewRef.current.wrappedInstance.send('setMarkers', props.searchMarkers.keywords, props.searchMarkers.options);
+			// TODO: remove this options hack when aceeditor is removed
+			// Currently the webviewRef will send out an ipcMessage to set the results count
+			// Also setting it here will start an infinite loop of repeating the search
+			// Unfortunately we can't remove the function in the webview setMarkers
+			// until the aceeditor is remove.
+			// The below search is more accurate than the webview based one as it searches
+			// the text and not rendered html (rendered html fails if there is a match
+			// in a katex block)
+			let options = { notFromAce: true };
+			if (props.searchMarkers.options) {
+				options = Object.assign(props.searchMarkers.options, options);
+			}
+			webviewRef.current.wrappedInstance.send('setMarkers', props.searchMarkers.keywords, options);
+			if (editorRef.current) {
+
+				const matches = editorRef.current.setMarkers(props.searchMarkers.keywords, props.searchMarkers.options);
+				props.setLocalSearchResultCount(matches);
+			}
```

Comment:
Thanks for adding the comment to explain the hack. Could you also add some keyword that can be searched to find the three instances where the code needs to be removed? Like "SEARCHMARKERHACK" for example, and maybe add the marker at the beginning and at the end of each related block. Also should the code be deleted at that time or replaced by something else?

Reply:
I've  wrapped each block in SEARCHHACK and provided instructions on what to do when aceeditor is removed.

Reply:
Perfect, thanks!

Reply:
I was just thinking about this today, maybe instead of being a hack I should add support here (backend only) for toggling between searching in editor and searching in viewer. That way it should be much easier to eventually add support from the frontend.